### PR TITLE
Add retry when data loading fails

### PR DESCRIPTION
**Pull request by jaymo001**
## Description 

1. When source data loading fails, wait and retry 2 times before fail the job. This could happen as the source is still being generated.
2. Fix test setup to enable Mockito Mock usage

## How was this PR tested?
Add unit test

## Does this PR introduce any user-facing changes?
No

### CODE REVIEW

File: feathr-impl/src/main/scala/com/linkedin/feathr/offline/source/dataloader/BatchDataLoader.scala [hunk 1]
```diff
@@ -87,12 +84,20 @@ private[offline] class BatchDataLoader(ss: SparkSession, location: DataLocation,
...
-        ss.read.format("csv").option("header", "true").option("delimiter", csvDelimiterOption).load(dataPath)
+      case _: Throwable =>
+        // If data loading from source failed, retry it automatically, as it might due to data source still being written into.
+        log.info(s"Loading ${location} failed, retrying for ${retry}-th time..")
+        if (retry > 0) {
+          Thread.sleep(DATA_LOAD_WAIT_IN_MS)
+          loadDataFrameWithRetry(dataIOParameters, jobConf, retry - 1)
+        } else {
+          // Throwing exception to avoid dataLoaderHandler hook exception from being swallowed.
+          throw new FeathrInputDataException(ErrorLabel.FEATHR_USER_ERROR, s"Failed to load ${dataPath} after ${MAX_DATA_LOAD_RETRY} retries.")
```
Comment (bozhonghu) [inline feathr-impl/src/main/scala/com/linkedin/feathr/offline/source/dataloader/BatchDataLoader.scala:95]: Does this swallow the original exception? I think we want to still show the original error for debugging purposes right?

File: feathr-impl/src/test/java/com/linkedin/feathr/common/AutoTensorizableTypesTest.java [hunk 2]
```diff
@@ -6,15 +6,14 @@
...
-import org.scalatest.testng.TestNGSuite;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import static java.util.Collections.*;
 import static org.testng.Assert.*;
 
 
-public class AutoTensorizableTypesTest extends TestNGSuite {
+public class AutoTensorizableTypesTest {
```
Comment (bozhonghu) [inline feathr-impl/src/test/java/com/linkedin/feathr/common/AutoTensorizableTypesTest.java:16]: Seems like there is a bunch of scala test related changed, can you update the description to reflect why the changes were made?

**Comment by xiaoyongzhu** [comment]
Thanks @jaymo001 ! can you merge main back to your branch just to make sure CI is OK? there are quite some bug fixes on main

**Review by bozhonghu — APPROVED** [review]
LGTM
